staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go

```diff
@@ -273,3 +275,44 @@ func createGetRequestWithUrl(rawUrlString string) *http.Request {
 		URL:    url,
 	}
 }
+
+func TestInformerSyncHealthChecker(t *testing.T) {
+	t.Run("test that check returns nil when all informers are started", func(t *testing.T) {
+		healthChecker := NewInformerSyncHealthz(sharedInformerFactoryStub{
+			startedByInformerType: map[reflect.Type]bool{
+				reflect.TypeOf(corev1.Pod{}): true,
+			},
+		})
+
+		err := healthChecker.Check(nil)
+		if err != nil {
+			t.Errorf("Got %v, expected no error", err)
+		}
+	})
+
+	t.Run("test that check returns err when there is not started informer", func(t *testing.T) {
+		healthChecker := NewInformerSyncHealthz(sharedInformerFactoryStub{
+			startedByInformerType: map[reflect.Type]bool{
+				reflect.TypeOf(corev1.Pod{}): true,
+				reflect.TypeOf(corev1.Service{}): false,
+				reflect.TypeOf(corev1.Node{}): true,
+			},
+		})
+
+		err := healthChecker.Check(nil)
+		if err == nil {
+			t.Errorf("expected error, got: %v", err)
+		}
+	})
+}
+
+type sharedInformerFactoryStub struct {
+	informers.SharedInformerFactory
```

Comment:
yeah i was just saying that if you defined the input interface in healthz.go with just the one method `WaitForCacheSync`, then you wouldn't need this. But it's not really a big deal.